1.0.184 might be causing a compilation error.

I got a compilation error which doesn't occur with `serde 1.0.183`. Is it a some breaking change?

[example project](https://github.com/tacogips/serde_184_error)

Cargo.toml
```
[package]
name = "serde_184_error"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
sentry = "0.31.5"

#The compile error not occured on serde = "= 1.0.183"
serde = "1.0.184"

```

Error message
```
    Updating crates.io index
    Checking sentry-types v0.31.5
error[E0507]: cannot move out of `*self` which is behind a shared reference
    --> /g/cargo/registry/src/index.crates.io-6f17d22bba15001f/sentry-types-0.31.5/src/protocol/v7.rs:1083:10
     |
1083 | #[derive(Serialize, Deserialize, Debug, Clone, Parti...
     |          ^^^^^^^^^
     |          |
     |          data moved here
     |          move occurs because `unrecognized` has type `v7::Context`, which does not implement the `Copy` trait
     |
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for moreinfo)
help: consider borrowing here
     |
1083 | #[derive(&Serialize, Deserialize, Debug, Clone, PartialEq)]
     |          +

For more information about this error, try `rustc --explain E0507`.
error: could not compile `sentry-types` (lib) due to previouserror
```

--- a/serde_derive/src/ser.rs
+++ b/serde_derive/src/ser.rs
@@ -409,9 +409,9 @@ fn serialize_enum(params: &Parameters, variants: &[Variant], cattrs: &attr::Cont
         })
         .collect();
 
-    if cattrs.non_exhaustive() {
+    if cattrs.remote().is_some() && cattrs.non_exhaustive() {
         arms.push(quote! {
-            unrecognized => _serde::__private::Err(_serde::ser::Error::custom(_serde::__private::ser::CannotSerializeVariant(unrecognized))),
+            ref unrecognized => _serde::__private::Err(_serde::ser::Error::custom(_serde::__private::ser::CannotSerializeVariant(unrecognized))),
         });
     }
 

--- a/test_suite/tests/test_remote.rs
+++ b/test_suite/tests/test_remote.rs
@@ -127,7 +127,7 @@ struct Test {
     enum_concrete: remote::EnumGeneric<u8>,
 
     #[serde(with = "ErrorKindDef")]
-    io_error_kind: std::io::ErrorKind,
+    io_error_kind: ErrorKind,
 }
 
 #[derive(Serialize, Deserialize)]
@@ -200,8 +200,16 @@ enum EnumConcrete {
     Variant(u8),
 }
 
+#[derive(Debug)]
+enum ErrorKind {
+    NotFound,
+    PermissionDenied,
+    #[allow(dead_code)]
+    ConnectionRefused,
+}
+
 #[derive(Serialize, Deserialize)]
-#[serde(remote = "std::io::ErrorKind")]
+#[serde(remote = "ErrorKind")]
 #[non_exhaustive]
 enum ErrorKindDef {
     NotFound,

